subprojects/platform-native/src/testFixtures/groovy/org/gradle/nativeplatform/fixtures/app/Swift3Test.groovy

```diff
@@ -23,12 +23,11 @@ class Swift3Test extends XCTestSourceFileElement {
 
     @Override
     List<XCTestCaseElement> getTestCases() {
-        return [testCase("testSwift3OnlyCode",
-            """typealias Name = (firstName: String, lastName: String)
-                let names: [Name] = [("Bart", "den Hollander")]
-                
-                names.forEach({ first, last in
-                    print(last)  // "den Hollander"
+        return [testCase("testOnlyOneEntryWithSpecificFirstAndLastName",
+            """// Assume only one entry
+                getNames().forEach({ first, last in
```

Comment:
I'd probably add some kind of verification that the test was compiled as swift 3 source, e.g. an assertion and `#if` or use some swift 3 only syntax in the test. Plus a similar thing for the swift 4 test source as well.